node/state_api/src/lib.rs

```diff
@@ -0,0 +1,397 @@
+                        all_faults.push(Fault {
+                            miner: *m,
+                            fault: fault_start,
+                        })
+                    }
+                    Ok(())
+                },
+            )
+        })
+        .collect::<Result<Vec<_>, String>>()?;
+    Ok(all_faults)
+}
+/// returns a bitfield indicating the recovering sectors of the given miner
+pub fn state_miner_recoveries<DB>(
+    state_manager: &StateManager<DB>,
+    actor: &Address,
+    key: &TipsetKeys,
+) -> Result<BitField, BoxError>
+where
+    DB: BlockStore,
+{
+    let tipset = ChainStore::new(state_manager.get_block_store()).tipset_from_keys(key)?;
+    state_manager::utils::get_miner_recoveries(&state_manager, &tipset, actor).map_err(|e| e.into())
+}
+
+/// returns the power of the indicated miner
+pub fn state_miner_power<DB>(
+    state_manager: &StateManager<DB>,
+    actor: &Address,
+    key: &TipsetKeys,
+) -> Result<(Option<Claim>, Claim), BoxError>
+where
+    DB: BlockStore,
+{
+    let tipset = ChainStore::new(state_manager.get_block_store()).tipset_from_keys(key)?;
+    state_manager::utils::get_power(&state_manager, &tipset, Some(actor)).map_err(|e| e.into())
+}
+
+pub fn state_page_collateral<DB>(
+    _state_manager: &StateManager<DB>,
+    _: &TipsetKeys,
+) -> Result<BigUint, BoxError>
+where
+    DB: BlockStore,
+{
+    Ok(BigUint::zero())
+}
+
+/// runs the given message and returns its result without any persisted changes.
+pub fn state_call<DB>(
+    state_manager: &StateManager<DB>,
+    message: &mut UnsignedMessage,
+    key: &TipsetKeys,
+) -> Result<InvocResult<UnsignedMessage>, BoxError>
+where
+    DB: BlockStore,
+{
+    let tipset = ChainStore::new(state_manager.get_block_store()).tipset_from_keys(key)?;
+    call::state_call(&state_manager, message, Some(tipset)).map_err(|e| e.into())
+}
+
+/// returns the result of executing the indicated message, assuming it was executed in the indicated tipset.
+pub fn state_reply<DB>(
+    state_manager: &StateManager<DB>,
+    key: &TipsetKeys,
+    cid: &Cid,
+) -> Result<InvocResult<UnsignedMessage>, BoxError>
+where
+    DB: BlockStore,
+{
+    let tipset = ChainStore::new(state_manager.get_block_store()).tipset_from_keys(key)?;
+    let (msg, ret) = call::state_replay(&state_manager, &tipset, cid)?;
+
+    Ok(InvocResult {
+        msg,
+        msg_rct: ret.msg_receipt().clone(),
+        actor_error: ret.act_error().map(|e| e.to_string()),
+    })
+}
+
+/// returns a state tree given a tipset
+pub fn state_for_ts<DB>(
+    state_manager: &StateManager<DB>,
+    maybe_tipset: Option<Tipset>,
+) -> Result<StateTree<DB>, BoxError>
+where
+    DB: BlockStore,
+{
+    let block_store = state_manager.get_block_store_ref();
+    let tipset = if maybe_tipset.is_none() {
+        chain::get_heaviest_tipset(block_store)?
+    } else {
+        maybe_tipset
+    };
+
+    let (st, _) = task::block_on(state_manager.tipset_state(&tipset.unwrap()))?;
```

Comment:
There doesn't seem to be any guarantee `get_heaviest_tipset` will be Some, maybe handle the error unless I'm missing something?